database/src/migrations/20240310163000_observation_sign.ts

```diff
@@ -0,0 +1,104 @@
+      description                                   varchar(256),
+      record_end_date                               timestamptz(6),
+      create_date                                   timestamptz(6)     DEFAULT now() NOT NULL,
+      create_user                                   integer            NOT NULL,
+      update_date                                   timestamptz(6),
+      update_user                                   integer,
+      revision_count                                integer            DEFAULT 0 NOT NULL,
+      CONSTRAINT observation_subcount_sign_id_pk PRIMARY KEY (observation_subcount_sign_id)
+    );
+
+    COMMENT ON TABLE observation_subcount_sign IS 'This table is intended to store options that users can select for the sign of a subcount.';
+    COMMENT ON COLUMN observation_subcount_sign.observation_subcount_sign_id IS 'Primary key for observation_subcount_sign.';
+    COMMENT ON COLUMN observation_subcount_sign.name IS 'Name of the sign option.';
+    COMMENT ON COLUMN observation_subcount_sign.description IS 'Description of the sign option.';
+    COMMENT ON COLUMN observation_subcount_sign.record_end_date IS 'End date of the sign option.';
+    COMMENT ON COLUMN observation_subcount_sign.create_date IS 'The datetime the record was created.';
+    COMMENT ON COLUMN observation_subcount_sign.create_user IS 'The id of the user who created the record as identified in the system user table.';
+    COMMENT ON COLUMN observation_subcount_sign.update_date IS 'The datetime the record was updated.';
+    COMMENT ON COLUMN observation_subcount_sign.update_user IS 'The id of the user who updated the record as identified in the system user table.';
+    COMMENT ON COLUMN observation_subcount_sign.revision_count IS 'Revision count used for concurrency control.';
+
+    ----------------------------------------------------------------------------------------
+    -- Add triggers for user data
+    ----------------------------------------------------------------------------------------
+    CREATE TRIGGER audit_observation_subcount_sign BEFORE INSERT OR UPDATE OR DELETE ON biohub.observation_subcount_sign FOR EACH ROW EXECUTE PROCEDURE tr_audit_trigger();
+    CREATE TRIGGER journal_observation_subcount_sign AFTER INSERT OR UPDATE OR DELETE ON biohub.observation_subcount_sign FOR EACH ROW EXECUTE PROCEDURE tr_journal_trigger();
+
+    ----------------------------------------------------------------------------------------
+    -- Modify observation_subcount table to include sign
+    ----------------------------------------------------------------------------------------
+    ALTER TABLE observation_subcount ADD COLUMN sign_id INTEGER;
+    COMMENT ON COLUMN observation_subcount.sign_id IS 'Foreign key referencing the sign id.';
+    ALTER TABLE observation_subcount ADD CONSTRAINT observation_subcount_sign_fk FOREIGN KEY (sign_id) REFERENCES observation_subcount_sign(observation_subcount_sign_id);
```

Comment:
Couple of things:

For foreign keys, can we keep the column names identical for both tables. Just helps with identifying which foreign column it points to if the names match.

`observation_subcount_sign_id`

--------

We'll want to add a foreign key index here as well.

`CREATE INDEX observation_subcount_idx2 ON observation_subcount(observation_subcount_sign_id);`

Foreign keys (in PSQL specifically) don't automatically generate an index, so we have to include it ourselves. Speeds up any queries that involve joining the subcount and sign tables.

--------

Can we change the name of the foreign key constraint to `observation_subcount_sign_fk1` (just adding the `1`).

Wont change anything, but the convention we use to date is to add a number to the end of any indexes or constraints, so if new ones get added we can just keep the same name and increment the number.

Reply:
Fixed 👍